chirp/drivers/retevis_ha1g.py

```diff
@@ -0,0 +1,3105 @@
+     char fastcall2[16];
+     char fastcall3[16];
+     char fastcall4[16];
+     char fastcall5[16];
+     char fastcall6[16];
+     char fastcall7[16];
+     char fastcall8[16];
+     char fastcall9[16];
+     char fastcall10[16];
+     u8 rev_1:6,
+        DecodingEnable:1,
+        EncodingEnable:1;
+     u8 dtmfstatus;
+     u8 rev_2[12];
+} dtmfs[4];
+"""
+
+
+class HandShakeStuts(Enum):
+    Normal = 0
+    Wrong = 1
+    PwdWrong = 3
+    RadioWrong = 4
+
+
+class Clone_TypeEnum(Enum):
+    radioHead = 2
+    radioInfo = 3
+    radioVer = 4
+    settingData = 6
+    zoneData = 7
+    channelData = 8
+    scanData = 11
+    vfoScanData = 12
+    alarmData = 13
+    dTMFData = 15
+    outFactoryData = 17
+
+
+class File_AddrEnum(Enum):
+    radioHead = 0
+    radioInfo = 0xE
+    radioVer = 0x52
+    settingData = 0x5C
+    zoneData = 0xC0
+    channelData = 0xD42
+    scanData = 0xB5C2
+    vfoScanData = 0xC3E4
+    alarmData = 0xC444
+    dTMFData = 0xC848
+    outFactoryData = 0xCCB4
+
+
+DTMFCHARSET = "0123456789ABCDabcd#*"
+NAMECHATSET = chirp_common.CHARSET_ALPHANUMERIC + "-/;,._!? *#@$%&+=/<>~(){}]'"
+SPECIAL_MEMORIES = {"VFOA": -2, "VFOB": -1}
+BANDWIDEH_LIST = ["NFM", "FM"]
+POWER_LEVELS = [
+    chirp_common.PowerLevel("Low", watts=0),
+    chirp_common.PowerLevel("High", watts=2),
+]
+TIMEOUTTIMER_LIST = [{"name": "%ss" % (x * 5), "id": x}
+                     for x in range(1, 64, 1)]
+TOTPERSMISSION_LIST = ["Always", "CTCSS/DCS Match",
+                       "Channel Free", "Receive Only"]
+SQUELCHLEVEL_LIST = ["AlwaysOpen", "1", "2", "3", "4", "5", "6", "7", "8", "9"]
+AUTOSCAN_LIST = ["OFF", "Auto Scan System"]
+OFFLINE_REVERSAL_LIST = ["OFF", "Freq Reversql", "Talk Around"]
+SIDE_KEY_LIST = [
+    {"name": "OFF", "id": 0},
+    {"name": "TX Power", "id": 1},
+    {"name": "Scan", "id": 2},
+    {"name": "FM Radio", "id": 3},
+    {"name": "Talkaround/Reversal", "id": 5},
+    {"name": "Monitor", "id": 15},
+    {"name": "Zone Plus", "id": 20},
+    {"name": "Zone Minus", "id": 21},
+    {"name": "Squelch", "id": 28},
+    {"name": "Emergency Start", "id": 29},
+    {"name": "Emergency Stop", "id": 30},
+    {"name": "Optional DTMF Code", "id": 32},
+    {"name": "Prog PTT", "id": 31},
+]
+Fre_Step_List = [
+    {"name": "2.5KHZ", "id": 2500},
+    {"name": "5KHZ", "id": 5000},
+    {"name": "6.25KHZ", "id": 6250},
+    {"name": "8.33KHz", "id": 8330},
+    {"name": "10KHZ", "id": 10000},
+    {"name": "12.5KHZ", "id": 12500},
+    {"name": "25KHZ", "id": 25000},
+]
+ALARM_LIST = [{"name": "OFF", "id": 255}]
+DTMFSYSTEM_LIST = [{"name": "OFF", "id": 15}]
+
+
+class HA1GBank(chirp_common.NamedBank):
+    """A HA1G bank"""
+
+    def get_name(self):
+        _zone_data = get_zone_index_list(self._model._radio)
+        _zone_list = self._model._radio._memobj.zones
+        if self.index in _zone_data:
+            zone_name = "".join(
+                filter(_zone_list[self.index].name, NAMECHATSET, 12)
+                )
```

Comment:
Please do not waste vertical space with close parens on single lines. This should be on the previous line.